### ISSUE TITLE
[Bug] SSL erron on click on "Switch Invidious Instance" in a onion instance

### ISSUE THREAD

**Issue by HELLoSKUuLL** [iv-org/invidious#2438]
**Describe the bug**
If using Tor Browser in a [onion url](http://grwp24hodrefzvjjuccrkw3mjq4tzhaaq32amf33dzpmuxe7ilepcmad.onion), the link `Switch Invidious Instance` gives an error

**Steps to Reproduce**
1. Access a onion url like [this](http://grwp24hodrefzvjjuccrkw3mjq4tzhaaq32amf33dzpmuxe7ilepcmad.onion)
2. Go to any video or channel
3. Click in `Switch Invidious Instance`

**Logs**
Title: `SSL_connect: error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED (OpenSSL::SSL::Error)`
Date: `2021-09-30T23:21:39Z`
Route: `/redirect?referer=%2Fwatch%3Fv%3D_o2y1SxprA0`
Version: `2021.09.28-82e6f6e0 @ master`
<details>
<summary>Backtrace</summary>
<p>
   
```
SSL_connect: error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED (OpenSSL::SSL::Error)
  from lib/lsquic/src/lsquic/patch.cr:174:11 in 'new:context:sync_close:hostname'
  from src/invidious/helpers/crystal_class_overrides.cr:34:5 in 'io'
  from /usr/share/crystal/src/http/client.cr:664:5 in 'exec_internal_single'
  from /usr/share/crystal/src/http/client.cr:587:5 in 'exec'
  from src/invidious/helpers/utils.cr:421:32 in '->'
  from lib/kemal/src/kemal/route.cr:12:9 in '->'
  from /usr/share/crystal/src/primitives.cr:266:3 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/static_file_handler.cr:189:11 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'process'
  from /usr/share/crystal/src/http/server.cr:500:5 in '->'
  from /usr/share/crystal/src/primitives.cr:266:3 in 'run'
  from ???
```
</p>
</details>

**Screenshots**

**Additional context**
The onion site has no SSL

**Comment by unixfox** [comment]
@syeopite Any ideas why it's doing that?

**Comment by syeopite** [comment]
No, and I'm quite frankly stumped on this. If I had to guess, it seems like the OpenSSL and LibCrypto patches from lsquic.cr, overwrote something from Crystal's internal bindings, breaking it's ability to query other domains without lsquic.cr when running through tor.

So, if we replaced this usage of the native crystal client:

https://github.com/iv-org/invidious/blob/cb525af0a28aa8e8d3394eb3074eee39ae0b594a/src/invidious/helpers/utils.cr#L415

With a QUIC client, like so:

```crystal
instance_api_client = make_client(URI.parse("https://api.invidious.io"))
```
 
We should be able to fix the instance redirection — and considering how the parts of Invidious that queries other services, which all uses either `make_client` or `YT_POOL`, functions without SSL errors, I'm fairly certain this is the cause of it.
